centreon/src/App/ResourceConfiguration/Infrastructure/Legacy/LegacyServiceCategoryPermissionVoter.php

```
@@ -0,0 +1,68 @@
+     */
+    private const LEGACY_PERMISSION_MAP = [
+        ServiceCategoryPermissionEnum::CanRead->value => Contact::ROLE_CONFIGURATION_SERVICES_CATEGORIES_READ,
+        ServiceCategoryPermissionEnum::CanWrite->value => Contact::ROLE_CONFIGURATION_SERVICES_CATEGORIES_READ_WRITE,
+    ];
+
+    protected function supports(string $attribute, mixed $subject): bool
+    {
+        return ServiceCategoryPermissionEnum::tryFrom($attribute) !== null;
+    }
+
+    protected function voteOnAttribute(string $attribute, mixed $subject, TokenInterface $token, ?Vote $vote = null): bool
```

PHPMD (2.15.0)

Avoid unused parameters such as '$subject'. (Unused Code Rules)

(UnusedFormalParameter)